### PR TITLE
Add serviceDiscovery apis

### PR DESCRIPTION
**Pull request by liuzhengbaihe**

### CODE REVIEW

File: src/manager/framework/api/http_api.go [hunk 1]
```diff
@@ -372,6 +406,104 @@ func (api *AppService) GetAppVersion(request *restful.Request, response *restful
...
+			serviceDiscovery := types.ServiceDiscovery{
+				TaskId:  slot.ID,
+				AppId:   slot.App.ID,
+				AppMode: string(slot.App.Mode),
+				IP:      slot.Ip,
+			}
+			serviceDiscoveries = append(serviceDiscoveries, serviceDiscovery)
+		}
+	} else {
+		for _, slot := range slots {
```
Comment (cmingxu) [inline src/manager/framework/api/http_api.go]: only RUNNING & healthy slot can be expose here

File: src/manager/framework/api/http_api.go [hunk 2]
```diff
@@ -372,6 +406,104 @@ func (api *AppService) GetAppVersion(request *restful.Request, response *restful
...
+			serviceDiscovery := types.ServiceDiscovery{
+				TaskId:  slot.ID,
+				AppId:   slot.App.ID,
+				AppMode: string(slot.App.Mode),
+				IP:      slot.AgentHostName,
+			}
+			serviceDiscovery.Url = strings.ToLower(strings.Replace(slot.ID, "-", ".", -1))
```
Comment (cmingxu) [inline src/manager/framework/api/http_api.go]: should add domain here. 

better way would be add a method for slot return domain

File: src/manager/framework/api/http_api.go [hunk 3]
```diff
@@ -372,6 +406,104 @@ func (api *AppService) GetAppVersion(request *restful.Request, response *restful
...
+	allServiceDiscoveries := make([]types.ServiceDiscovery, 0)
+	for _, app := range api.Scheduler.ListApps(appFilterOptions) {
+		serviceDiscoveries := getServiceDiscoveris(app)
+		allServiceDiscoveries = append(allServiceDiscoveries, serviceDiscoveries...)
+	}
+	response.WriteEntity(allServiceDiscoveries)
+}
+
+func (api *AppService) GetAllServiceDiscoveriesMD5(request *restful.Request, response *restful.Response) {
+	appFilterOptions := scheduler.AppFilterOptions{}
```
Comment (cmingxu) [inline src/manager/framework/api/http_api.go]: how about move filteroptions struct under types

File: src/types/service-discovery.go [hunk 4]
```diff
@@ -0,0 +1,17 @@
+package types
+
+type ServiceDiscovery struct {
+	TaskId           string             `json:"taskId"`
```
Comment (xychu) [inline src/types/service-discovery.go]: Better to use TaskID instead of TaskId and taskID for taskId.

**Comment by xychu** [comment]
+1

**Pull request merged**